### DISCUSSION TITLE
Pushing arbitrary blobs

### DISCUSSION THREAD

**Discussion by vsoch** [supercontainers/containers-wg#3] [Ideas]
This is a discussion thread for the [Pushing Arbitrary Blobs](https://supercontainers.github.io/containers-wg/ideas/arbitrary-blobs/) idea. You can:

 - vote this discussion up to express interest in the idea
 - add comments, feedback, or ideas for discussion in this thread.

**Comment by dtrudg** [comment]
> A new project (CNCF or OCI)
> Given that we have support for pushing arbitrary blobs, we could easy create an OCI compliant Singularity Registry.

I tend to think this point is covered already? Harbor is a CNCF project and Singularity can push to/pull from Harbor today via `singularity push/pull oras://`

**Comment by dtrudg** [comment]
> What are the challenges that we are trying to solve?
>    Any OCI registry should be able to support custom content types.

A fundamental challenge here will be one of incentives. For open-source self-hosted, or cloud provider supported private registries, there is probably a lot of mileage in submitting PRs / lobbying for opening contentType restrictions.

For the public container registries I'm not sure we'll get there. E.g. DockerHub is the most visible registry for OCI containers. However, I don't think there is any incentive for Docker the company to provide free hosting for e.g Singularity container images, which cannot be used with their products? Arbitrary blob-only uploads are probably even less likely to happen - likely provides a trivial way to share pirated music / videos etc.

Maybe a wish that "Any self-hosted / private OCI registry should be configurable to allow arbitrary custom contentTypes" would be a better scope for the aspiration here?

**Reply by vsoch** [reply]
@dtrudg I think that's already possible with artifacts/custom content types? This particular issue is probably a bit dated in that it's possible but the challenges remaining (as you noted) are more about convincing people to do it. I agree that Docker Hub is not going to allowing pushing Singularity, but maybe they would if people paid for it?
